api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricsServiceLifecycle.java

```diff
@@ -0,0 +1,241 @@
+import com.google.common.util.concurrent.Uninterruptibles;
+
+/**
+ * Bean created on startup to manage the lifecycle of the {@link MetricsService} instance shared in application scope.
+ *
+ * @author John Sanda
+ * @author Thomas Segismont
+ */
+@ApplicationScoped
+@Eager
+public class MetricsServiceLifecycle {
+    private static final Logger LOG = LoggerFactory.getLogger(MetricsServiceLifecycle.class);
+
+    /**
+     * @see #getState()
+     */
+    public enum State {
+        STARTING, STARTED, STOPPING, STOPPED, FAILED
+    }
+
+    private final MetricsServiceImpl metricsService;
+    private final ScheduledExecutorService lifecycleExecutor;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(CASSANDRA_CQL_PORT)
+    private String cqlPort;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(CASSANDRA_NODES)
+    private String nodes;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(CASSANDRA_KEYSPACE)
+    private String keyspace;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(CASSANDRA_RESETDB)
+    private String resetDb;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(WAIT_FOR_SERVICE)
+    private String waitForService;
+
+    private volatile State state;
+    private int connectionAttempts;
+    private Session session;
+
+    MetricsServiceLifecycle() {
+        // Create the shared instance now and initialize it with the C* session when ready
+        metricsService = new MetricsServiceImpl();
+        ThreadFactory threadFactory = r -> {
+            Thread thread = Executors.defaultThreadFactory().newThread(r);
+            thread.setName(MetricsService.class.getSimpleName().toLowerCase(Locale.ROOT) + "-lifecycle-thread");
+            return thread;
+        };
+        // All lifecycle operations will be executed on a single thread to avoid synchronization issues
+        lifecycleExecutor = Executors.newSingleThreadScheduledExecutor(threadFactory);
+        state = State.STARTING;
+    }
+
+    /**
+     * Returns the lifecycle state of the {@link MetricsService} shared in application scope.
+     *
+     * @return lifecycle state of the shared {@link MetricsService}
+     */
+    public State getState() {
+        return state;
+    }
+
+    @PostConstruct
+    void init() {
+        lifecycleExecutor.submit(this::startMetricsService);
+        if (Boolean.parseBoolean(waitForService)
+            // "hawkular-metrics.backend" is not a real Metrics configuration parameter (there's a single
+            // MetricsService implementation, which is backed by Cassandra).
+            // But it's been used historically to wait for the service to be available before completing the deployment.
+            // Therefore, we still use it here for backward compatibililty.
+            // TODO remove when Hawkular build has been updated to use the eager startup flag
+            || "embedded_cass".equals(System.getProperty("hawkular-metrics.backend"))) {
```

Comment:
When we started embedding Cassandra as an EAR, one of the issues we faced was with start up and initialization. It was possible for metrics to start up before Cassandra was initialized and ready to accept client connections. Checking for the flag here was done to let us know that we should check to see if Cassandra has initialized. Because we now support hawkular starting before Cassandra, I think we need the initialization logic regardless of whether or not Cassandra is embedded. Can we just altogether remove this check?